### DISCUSSION TITLE
Replication

### DISCUSSION THREAD

**Discussion by KuDeSnik33ra** [erthink/libmdbx#331] [Q&A] [answered]
Are there any functions available for database replication? How can I get changes in DB since some time point in the past?

**Comment by erthink** [comment, marked as answer]
Basically, no.
MDBX (as historically inheriting a minimalism from LMDB) doesn't have/using [WAL](https://en.wikipedia.org/wiki/Write-ahead_logging), so no log of changes is available.

You should implement your own changelog or take look to libfpta, where an implementation of content-based synchronization/replication will be done this year, i.e. like the [RFC-4355](https://www.rfc-editor.org/rfc/rfc4533.txt).

**Comment by erthink** [comment]
To complete the picture, it should be noted that it is usable (successfully operated for several years) a scheme with master-slave replication at the block/storage device level of the database file.

This scheme is reliable provided that the file system is read-only on replicas and the block replication provider executes it without violating the write order on the master. For instance, the [DRBD (Distributed Replicated Block Device)](https://en.wikipedia.org/wiki/DRBD) cold be used for this.
